src/client/content_cao.cpp

```diff
@@ -1432,6 +1449,17 @@ void GenericCAO::updateBonePosition()
 			bone->updateAbsolutePosition();
 		}
 	}
+	// The following is needed for set_bone_pos to propagate to
+	// attached objects correctly.
+	// Irrlicht ought to do this, but doesn't when using EJUOR_CONTROL.
+	for (u32 i = 0; i < m_animated_meshnode->getJointCount(); ++i) {
+		auto bone = m_animated_meshnode->getJointNode(i);
+		// Look for the root bone.
+		if (bone != NULL && bone->getParent() == m_animated_meshnode) {
```

Comment:
This is almost line for line what irrlicht does in CAnimatedMeshSceneNode, in its EJUOR_READ path. Possibly by mistake, the other code paths do not do this, and some lines above we've just set EJUOR_CONTROL so we're sure to miss it.